lib/versioning/loose/generic.ts

```diff
@@ -39,41 +49,37 @@ export abstract class GenericVersioningApi<
     const length = Math.max(left.release.length, right.release.length);
     for (let i = 0; i < length; i += 1) {
       // 2.1 and 2.1.0 are equivalent
-      const part1 = left.release[i] ?? 0;
-      const part2 = right.release[i] ?? 0;
-      if (part1 !== part2) {
-        return part1 - part2;
+      const leftPart = left.release[i] ?? 0;
+      const rightPart = right.release[i] ?? 0;
+      if (leftPart < rightPart) {
+        return -1;
+      } else if (leftPart > rightPart) {
+        return 1;
```

Comment:
Every implementation based on `loose` uses it's own `_compare`, so sticking to the logic from `index.ts` (which is being leveraged in our buildpack repos)